tests/spread/charms/bundle/task.yaml

```diff
@@ -0,0 +1,23 @@
+summary: Recursively pack a bundle with all charms on
+manual: true
+systems:
+  - ubuntu-22.04-64
+kill-timeout: 30m
+
+environment:
+  BUNDLE/notebook_operators: https://github.com/canonical/notebook-operators
+
+prepare: |
+  git clone --depth=1 "${BUNDLE}" bundle
+
+restore: |
+  pushd bundle
+  charmcraft clean
+  popd
+
+  rm -rf bundle
+
+execute: |
+  cd bundle
+  charmcraft pack --verbose --include-all-charms
+  juju deploy --dry-run *.zip
```

Comment:
I don't know juju so how can we verify that the bundle contains the expected multiple charms?

Reply:
The bundle itself won't contain the charms — they'll be in the working directory and referenced (by their full path) from the bundle zip file. I've added some tests to ensure correctness. (Not sure how to make a parametrizable test like this with an explicit `--include-charms` so there's no actual e2e test for that at this point.)